code/modules/blob/blob.dm

```diff
@@ -4,23 +4,87 @@
 	icon = 'icons/mob/blob.dmi'
 	icon_state = "blob"
 	light_range = 3
+	light_color = "#b5ff5b"
 	desc = "Some blob creature thingy"
 	density = 1
 	opacity = 0
 	anchored = 1
 	mouse_opacity = 2
+	layer = 4
 
 	var/maxHealth = 30
 	var/health
+	var/regen_rate = 5
 	var/brute_resist = 4
+	var/laser_resist = 4 // Special resist for laser based weapons - Emitters or handheld energy weaponry. Damage is divided by this and THEN by fire_resist.
 	var/fire_resist = 1
+	var/secondary_core_growth_chance = 10.0 //% chance to grow a secondary blob core instead of whatever was suposed to grown. Secondary cores are considerably weaker, but still nasty.
 	var/expandType = /obj/effect/blob
+	var/obj/effect/blob/core/parent_core = null
+	var/growth_range = 0
+	var/blob_may_process = 1
+	var/hangry = 0 //if the blob will attack or not.
 
 /obj/effect/blob/New(loc)
+	processing_objects.Add(src)
 	health = maxHealth
 	update_icon()
 	return ..(loc)
 
+/obj/effect/blob/Destroy()
+	processing_objects.Remove(src)
+	..()
+
+/obj/effect/blob/process()
+	if(!parent_core)
+		src.take_damage(5)
+		src.regen_rate = -5
+		src.growth_range = 0
+		playsound(loc, 'sound/effects/splat.ogg', 50, 1)
+		return
+
+	if(prob(70))
+		for(var/mob/living/L in src.loc)
+			if(L.stat == DEAD)
+				if(prob(10))
+					L.gib()
+					if(health < maxHealth)
+						health += rand(10,30)
+						if(health > maxHealth)
+							health = maxHealth
+				continue
+			L.visible_message("<span class='danger'>The blob absorbs \the [L]!</span>", "<span class='danger'>The blob absorbs you!</span>")
+			playsound(loc, 'sound/effects/attackblob.ogg', 50, 1)
+			L.take_organ_damage(rand(5, 10))
+			if(health < maxHealth)
+				health += rand(1,10)
+				if(health > maxHealth)
+					health = maxHealth
+			hangry += 16
+
+		for(var/mob/living/L in range(src,"3x3"))
```

Comment:
what does this 3x3 thing do?
Is that a 3x3 square centred on the blob? if so thats no different from a range of 1